examples/demo/modules/blog/handler/blog_handler.e

```diff
@@ -120,153 +122,150 @@ feature -- Query
 			end
 		end
 
-	total_entries : NATURAL_32
-			-- Returns the number of total entries/posts
+	total_entries: NATURAL_32
+			-- Total number of entries/posts.
 		do
-			Result := node_api.blogs_count.to_natural_32
+			Result := blog_api.blogs_count.to_natural_32
 		end
 
 feature -- HTML Output
 
-	main_content_html (page: CMS_RESPONSE) : STRING
-			-- Return the content of the page as a html string
+	frozen main_content_html (page: CMS_RESPONSE): STRING
+			-- Content of the page as a html string.
+		do
+			create Result.make_empty
+			append_main_content_html_to (page, Result)
+		end
+
+	append_main_content_html_to (page: CMS_RESPONSE; a_output: STRING)
```

Comment:
As an additional comment, 
a_output.append ("abc" + a_string + "def")
should be replaced by 
a_output.append ("abc")
a_output.append (a_string)
a_output.append ("def")
to avoid intermediate string object creation.

BUT ... sometime this is not that critical; and we favor the readability of the code.